storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
The `:=` behavior you mean only applies to the new variable in this case. The `err` gets assigned a new value but does no shadowing happens. Still always confusing me.

https://golang.org/ref/spec#Short_variable_declarations